opendbc/car/hyundai/interface.py

```diff
@@ -25,7 +25,17 @@ def _get_params(ret: structs.CarParams, candidate, fingerprint, car_fw, experime
     # FIXME: the Optima Hybrid 2017 uses a different SCC12 checksum
     ret.dashcamOnly = candidate in {CAR.KIA_OPTIMA_H, }
 
-    hda2 = Ecu.adas in [fw.ecu for fw in car_fw]
+    # Check if the ADAS Driving ECU is present in the firmware response.
+    # If not found, check the CAM CAN bus fingerprint for known HDA2 steering messages (0x50 or 0x110),
+    # that indicate an HDA2 car even without a direct ADAS Driving ECU response.
+    if Ecu.adas in [fw.ecu for fw in car_fw]:
+      hda2 = True
+    else:
+      # Some HDA2 cars do not respond ADAS Driving ECU if the known HDA2 steering messages are present.
+      # The hda2 flag to initialize CanBus is not critical here since we only care about CAM bus steering messages.
```

Comment:
I don't think we've ever verified what is sending LFA here btw. The ADAS could actually exist but behind a new gateway (a FW scan on OBD port should show it), or it could actually not exist and radar might take over this function.